src/services/goals.js

```diff
@@ -347,6 +347,17 @@ export async function goalsByIds(id) {
             ],
             required: false,
           },
+          {
+            model: ActivityReportObjective,
+            as: 'activityReportObjectives',
+            attributes: [
+              'ttaProvided',
+            ],
+            required: false,
+            where: {
+              activityReportId,
+            },
+          },
```

Comment:
Goals link to ARs directly through the ActivityReportGoals table as well

Reply:
This join isn't to get the AR. Its to makes sure we get the associated 'ActivityReportObjectives' associated with the 'objective'. If the user deletes the 'objective' we need to clean these up as well.

Reply:
If the delete all objectives linking a goal to the AR, the ActivityReportGoals link should also be cleaned up.